### DISCUSSION TITLE
Is this a proper way to delect disconnect inside a loop (socketio.AsyncNamespace)

### DISCUSSION THREAD

**Discussion by 4n1qz5skwv** [miguelgrinberg/python-socketio#1171] [Q&A] [answered]
```python
class IndexSio(socketio.AsyncNamespace):
    async def on_connect(self, sid, *_, **__):
        ...

    async def on_disconnect(self, sid, *_, **__):
        ...

    async def on_start(self, sid):
        while True:
            try:
                ddd = await self.get_session(sid)
            except KeyError:
                # Is this a safe place to assume that client has disconnected ?
                break
            print(f'{ddd=}')
```

**Hint:** _the comment (except block)_

**Comment by miguelgrinberg** [comment, marked as answer]
The `disconnect` event is invoked when a client disconnects. There is no need for you to repeatedly get the session just to see if that errors out. For example, you can use an `Event` object in your `on_start` event handler for each client, which you signal from the disconnect handler.

**Reply by 4n1qz5skwv** [reply]
Thank you. It helped 👍

May be in future, we will have a built-in property, something like `self.is_connected` 🤞
